server/src/main/java/org/opensearch/common/settings/ClusterSettings.java

```diff
@@ -758,7 +759,12 @@ public void apply(Settings value, Settings current, Settings previous) {
                 SearchService.CLUSTER_ALLOW_DERIVED_FIELD_SETTING,
 
                 // Composite index settings
-                CompositeIndexSettings.STAR_TREE_INDEX_ENABLED_SETTING
+                CompositeIndexSettings.STAR_TREE_INDEX_ENABLED_SETTING,
+
+                // QueryGroup settings
+                QueryGroupServiceSettings.MAX_QUERY_GROUP_COUNT,
+                QueryGroupServiceSettings.NODE_LEVEL_REJECTION_THRESHOLD,
+                QueryGroupServiceSettings.NODE_LEVEL_CANCELLATION_THRESHOLD
```

Comment:
I think we should add resource specific settings as CPU and Memory are quite different in their behavior.